x-pack/plugins/security_solution/public/common/components/endpoint/agent_status.tsx

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiBadge } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { HostStatus } from '../../../../common/endpoint/types';
+import { HOST_STATUS_TO_BADGE_COLOR } from '../../../management/pages/endpoint_hosts/view/host_constants';
+import { Maybe } from '../../../../../observability/common/typings';
+
+export const AgentStatus = React.memo(({ hostStatus }: { hostStatus: Maybe<HostStatus> }) => {
```

Comment:
nice! 
I will refactor the endpoint list/details to also use this common component